File: app/src/main/java/com/niiadotei/storehouse/ui/supplies/SuppliesActivity.java
package com.niiadotei.storehouse.ui.supplies;

import android.os.Bundle;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.niiadotei.storehouse.data.DatabaseHelper;
import com.niiadotei.storehouse.databinding.ActivitySuppliesBinding;

public class SuppliesActivity extends AppCompatActivity {
    RecyclerView recyclerView;

    SuppliesViewModel suppliesViewModel;

    DatabaseHelper databaseHelper;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivitySuppliesBinding binding = ActivitySuppliesBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setTitle("Supplies History");

        ActionBar actionBar = getSupportActionBar();
        assert actionBar != null;
        actionBar.setDisplayHomeAsUpEnabled(true);

        recyclerView = binding.suppliesRecyclerView;

        databaseHelper = new DatabaseHelper(getApplicationContext());

        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
        suppliesViewModel = new SuppliesViewModel(this, databaseHelper.getSupplyChainArray());
        recyclerView.setAdapter(suppliesViewModel);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            this.finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/niiadotei/storehouse/ui/supplies/SuppliesActivity.kt
package com.niiadotei.storehouse.ui.supplies

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.niiadotei.storehouse.data.DatabaseHelper
import com.niiadotei.storehouse.databinding.ActivitySuppliesBinding

class SuppliesActivity : AppCompatActivity() {
    private lateinit var databaseHelper: DatabaseHelper

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivitySuppliesBinding.inflate(layoutInflater)
        setContentView(binding.root)

        title = "Supplies History"

        val actionBar = supportActionBar!!
        actionBar.setDisplayHomeAsUpEnabled(true)

        val recyclerView = binding.suppliesRecyclerView

        databaseHelper = DatabaseHelper(applicationContext)

        recyclerView.layoutManager = LinearLayoutManager(applicationContext)

        val suppliesViewModel = SuppliesViewModel(databaseHelper.supplyChainArray)
        recyclerView.adapter = suppliesViewModel
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }

        return super.onOptionsItemSelected(item)
    }
}